core/src/main/java/software/amazon/awssdk/core/util/Mimetypes.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright 2010-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.core.util;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.StringTokenizer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.utils.IoUtils;
+
+/**
+ * Utility class that maintains a listing of known Mimetypes, and determines the
+ * mimetype of files based on file extensions.
+ * <p>
+ * This class is obtained with the {#link {@link #getInstance()} method that
+ * recognizes loaded mime types from the file <code>mime.types</code> if this
+ * file is available at the root of the classpath. The mime.types file format,
+ * and most of the content, is taken from the Apache HTTP server's mime.types
+ * file.
+ * <p>
+ * The format for mime type setting documents is:
+ * <code>mimetype + extension (+ extension)*</code>. Any
+ * blank lines in the file are ignored, as are lines starting with
+ * <code>#</code> which are considered comments. Lines that have a mimetype but
+ * no associated extensions are also ignored.
+ */
+public final class Mimetypes {
+
+    /** The default XML mimetype: application/xml */
+    public static final String MIMETYPE_XML = "application/xml";
+
+    /** The default HTML mimetype: text/html */
+    public static final String MIMETYPE_HTML = "text/html";
+
+    /** The default binary mimetype: application/octet-stream */
+    public static final String MIMETYPE_OCTET_STREAM = "application/octet-stream";
+
+    /** The default gzip mimetype: application/x-gzip */
+    public static final String MIMETYPE_GZIP = "application/x-gzip";
+
+    public static final String MIMETYPE_TEXT_PLAIN = "text/plain";
+
+    private static final Logger LOG = LoggerFactory.getLogger(Mimetypes.class);
+
+    private static final String MIME_TYPE_PATH = "software/amazon/awssdk/core/util/mime.types";
+
+    private static final ClassLoader CLASS_LOADER = Mimetypes.class.getClassLoader();
```

Comment:
This may return null if the SDK is loaded by the bootstrap classloader for whatever reason. I think Kyle added a utility to get either the class loader or system class loader.